src/Sparrow/Platform/Posix/Syscall.cs

```diff
@@ -275,12 +276,34 @@ public static int AllocateFileSpace(int fd, long size, string file, out bool usi
                         // fallocate is not supported, we'll use lseek instead
                         usingWrite = true;
                         byte b = 0;
-                        if (pwrite(fd, &b, 1, size - 1) != 1L)
+                        int cifsRetry = 3;
+                        while (true)
```

Comment:
You can't do a `while(true)`

Reply:
I'll write `static int pwriteWithRetriesOrThrow()` for both cifs retries and EINTR w/wo timeouts

Reply:
ok